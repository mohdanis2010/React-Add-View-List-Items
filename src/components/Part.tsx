import React from "react";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import "../css/list-style.css";

const Part = ({
  name,
  id,
  status,
}: {
  name: string;
  id: number;
  status: string;
}) => (
  <List style={{ listStyle: "none" }} className="list-item">
    <ListItem>
      {" "}
      <ListItemText>
        Name: <span className="meta">{name}</span>
      </ListItemText>
    </ListItem>
    <ListItem>
      {" "}
      <ListItemText>
        ID: <span className="meta">{id}</span>
      </ListItemText>
    </ListItem>
    <ListItem>
      {" "}
      <ListItemText>
        Status: <span className="meta">{status}</span>
      </ListItemText>
    </ListItem>
  </List>
);

export default Part;
