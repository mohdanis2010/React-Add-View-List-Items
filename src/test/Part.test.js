import React from "react";
import { render, cleanup, getByTestId } from "@testing-library/react";
import Part from '../components/Part';

afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(<Part name="Hello!" status="status" id="1" />);
    expect(asFragment()).toMatchSnapshot();
  });

  // it('should equal to 0', () => {
  //   const { getByTestId } = render(<Part />); 
  //   expect(getByTestId('list-item')).toHaveClass('list-item')
  //  });




