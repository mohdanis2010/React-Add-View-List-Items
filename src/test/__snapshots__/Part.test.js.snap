// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="list-item"
  >
    <div
      class="tools"
    >
      Name: 
      <span
        class="meta"
      >
        Hello!
      </span>
    </div>
    <div
      class="tools"
    >
      ID: 
      <span
        class="meta"
      >
        1
      </span>
    </div>
    <div
      class="tools"
    >
      Status: 
      <span
        class="meta"
      >
        status
      </span>
    </div>
  </div>
</DocumentFragment>
`;
